plugins/module_utils/foreman_helper.py

```diff
@@ -26,6 +26,31 @@
     APYPIE_IMP_ERR = traceback.format_exc()
 
 
+def _entity_spec_helper(spec):
+    """Extend an entity spec by adding entries for all alt_ and flat_names.
+    Extract ansible compatible argument_spec on the way.
+    """
+    entity_spec = {'id': {}}
+    argument_spec = {}
+    for key, value in spec.items():
+        entity_value = {}
+        argument_value = value.copy()
+        if 'flat_name' in argument_value:
+            flat_name = argument_value.pop('flat_name')
+            entity_value['flat_name'] = flat_name
+            entity_spec[flat_name] = {}
+        if argument_value.get('type') == 'entity':
```

Comment:
Maybe a newline for readability?
```suggestion

        if argument_value.get('type') == 'entity':
```